### PR TITLE
feat: redis sink using depot

### PR DESCRIPTION
**Pull request by sumitaich1998**
We are extracting out Redis sink code from firehose and using the Redis sink connector from ODPF Depot library.

Here are the changed configs - 

- we are replacing INPUT_SCHEMA_PROTO_TO_COLUMN_MAPPING  config with the config SINK_REDIS_HASHSET_FIELD_TO_COLUMN_MAPPING . The Proto field index in the mapping must be replaced by the proto field name. 
- we are replacing SINK_REDIS_LIST_DATA_PROTO_INDEX  config with the config SINK_REDIS_LIST_DATA_FIELD_NAME . The Proto field index must be replaced by the proto field name
- we are replacing SINK_REDIS_LIST_DATA_PROTO_INDEX  config with the config SINK_REDIS_LIST_DATA_FIELD_NAME . The Proto field index must be replaced by the proto field name. 
- SINK_REDIS_KEY_TEMPLATE will now accept proto field names as the argument instead of the proto index.

### CODE REVIEW

File: src/main/java/io/odpf/firehose/sink/SinkFactory.java [hunk 1]
```diff
@@ -67,6 +70,16 @@ public void init() {
...
                 logSinkFactory.init();
                 return;
+            case REDIS:
+                redisSinkFactory = new RedisSinkFactory(
+                        ConfigFactory.create(RedisSinkConfig.class, config),
+                        statsDReporter);
+                try {
+                    redisSinkFactory.init();
+                } catch (IOException e) {
+                    throw new IllegalArgumentException("Exception occurred while creating sink", e);
```
Comment (lavkesh) [inline src/main/java/io/odpf/firehose/sink/SinkFactory.java:80]: ah. no. We should not catch it. 
The init() itself should throw RuntimeException.

Comment (sumitaich1998) [reply src/main/java/io/odpf/firehose/sink/SinkFactory.java:80]: it throws IOException, we'll need to catch and throw IllegalArgumentException in depot then

Comment (sumitaich1998) [reply src/main/java/io/odpf/firehose/sink/SinkFactory.java]: done

**Review by kevinbheda — APPROVED** [review]
LGTM

**Comment by kevinbheda** [comment]
dont need the prom sink commit in this branch, please remove it.

**Pull request merged**